auction-server/src/auction.rs

```diff
@@ -1589,34 +1601,36 @@ impl ChainStore for ChainStoreSvm {
             .try_into()
             .map_err(|_| anyhow!("Invalid svm signature"))?;
         if bids.len() != 1 {
-            return Err(anyhow!("Invalid number of bids: {}", bids.len()));
+            tracing::warn!(tx_hash = ?tx_hash, bids = ?bids, "multiple bids found for transaction hash");
         }
 
         let status = self
             .client
             .get_signature_status_with_commitment(&tx_hash, CommitmentConfig::confirmed())
             .await?;
 
-        match status {
-            Some(res) => Ok(Some(vec![match res {
+        let status = match status {
+            Some(res) => match res {
                 Ok(()) => BidStatusSvm::Won { result: tx_hash },
                 Err(_) => BidStatusSvm::Lost {
                     result: Some(tx_hash),
                 },
-            }])),
+            },
             None => {
                 // not yet confirmed
                 // TODO Use the correct version of the expiration algorithm, which is:
                 // the tx is not expired as long as the block hash is still recent.
                 // Assuming a certain block time, the two minute threshold is good enough but in some cases, it's not correct.
                 if bids[0].initiation_time + BID_MAXIMUM_LIFE_TIME_SVM < OffsetDateTime::now_utc() {
```

Comment:
the expiration of bids will depend on their individual initiation time, not on another bid's initiation time. so this should actually be a loop? e.g. it's possible the first of the bids gets processed but doesn't get confirmed but a later one does get confirmed?